zerver/lib/presence.py

```diff
@@ -112,22 +88,35 @@ def get_legacy_user_info(
         timestamp=most_recent_info["timestamp"],
     )
 
-    # Build a dictionary of client -> info.  There should
-    # only be one row per client, but to be on the safe side,
-    # we always overwrite with rows that are later in our list.
-    for info in info_rows:
-        result[info["client"]] = info
+    result[client_name] = most_recent_info
 
     return result
 
 
+def format_legacy_presence_dict(presence: UserPresence) -> Dict[str, Any]:
+    """
+    This function assumes it's being called right after the presence object was updated,
+    and is not meant to be used on old presence data.
+    """
+    if (
+        presence.last_active_time + datetime.timedelta(minutes=1, seconds=10)
```

Comment:
This constant should probably be extracted to a value in `computed_settings.py` as well.